shared/services/basket-service.ts

```diff
@@ -64,9 +64,13 @@ export class BasketService {
     }
 
     getExistingItem(item: BasketItem): OrderLineItem {
-        return this.items.find(x =>
-            x.name === item.name &&
-            x.version === item.version
+        let sortedItemOptions = item.options.sort();
+        return this.items.find(x => {
+            return x.name === item.name &&
+                x.version === item.version &&
+                x.options.length === sortedItemOptions.length &&
+                x.options.sort().every((option, i) => option === sortedItemOptions[i])
```

Comment:
`.sort()` modifies the original array
Another way would be that instead of the above 2 lines you could do this:
`x.options.slice().sort().join() === item.options.slice().sort().join()` 

It would be good to have a unit test around this as we can easily unit test the shared code.